src/components/Explore/helpers/doesQueryMatchDataFrameLabels.ts

```diff
@@ -0,0 +1,12 @@
+import { DataFrame } from '@grafana/data';
+
+export const doesQueryMatchDataFrameLabels = (searchQuery?: string) => (dataFrame: DataFrame) => {
+  const pattern = searchQuery?.trim();
+  if (!pattern) {
+    return true;
+  }
+
+  const regex = new RegExp(pattern, 'i');
```

Comment:
That would allow to search using regular expression which is kind of cool, but I was wonder if:
- should we be more explicit about it (e.g. in the prompt) + add tests coverage for it?, or
- or lowercase label+input and do simple case-insensitive search?

Reply:
Good question, I believe it all depends how we want this feature to evolve. IMHO, I'd love to see all apps going towards what (e.g.) VSCode provides:

<img width="328" alt="image" src="https://github.com/user-attachments/assets/0a99abfb-8cf6-4733-96ff-02bf85e8dd77">

I guess for now, the PR is good enough and we can discuss about it in the future.